src/sentry/middleware/ratelimit.py

```diff
@@ -8,13 +8,35 @@
 from django.utils.deprecation import MiddlewareMixin
 from rest_framework.request import Request
 
-from sentry.api.helpers.group_index.index import EndpointFunction, build_rate_limit_key
+from sentry.api.helpers.group_index.index import EndpointFunction
 from sentry.app import ratelimiter
 
 
 def get_rate_limit_key(view_func: EndpointFunction, request: Request):
     """Construct a consistent global rate limit key using the arguments provided"""
-    return build_rate_limit_key(view_func, request)
+
+    view = f"{view_func.__module__}.{view_func.__name__}"
+    http_method = request.method
+
+    request_user = getattr(request, "user", None)
+    user_id = getattr(request_user, "id", None)
+    is_sentry_app = getattr(request_user, "is_sentry_app", None)
+
+    request_access = getattr(request, "access", None)
+    org_id = getattr(request_access, "organization_id", None)
+
+    # Default to using an IP based ratelimit
+    if is_sentry_app and org_id:
+        category = "org"
+        id = org_id
+    elif user_id:
+        category = "user"
+        id = user_id
+    else:
+        category = "ip"
+        id = request.META["REMOTE_ADDR"]
```

Comment:
If for some reason `REMOTE_ADDR` is not in `request.META` this will result in the request failing. We should return a key of None, and have the rate limited ignore rate limits if this happens for now.